### PR TITLE
[engine] prefer Control to throw

### PR DESCRIPTION
**Pull request by nickchapman-da**
- prefer `Control` to `throw` for `interpretation.Error`
- kill `SEDamlException`

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala [hunk 1]
```diff
@@ -1033,19 +1039,24 @@ private[lf] object SBuiltin {
...
-          throw convTxError(err)
+          Control.Error(convTxError(err))
       }
     }
   }
 
   // SBCastAnyContract: ContractId templateId -> Any -> templateId
-  final case class SBCastAnyContract(templateId: TypeConName) extends SBuiltinPure(2) {
-    override private[speedy] def executePure(args: util.ArrayList[SValue]) = {
+  final case class SBCastAnyContract(templateId: TypeConName) extends SBuiltin(2) {
```
Comment (carlpulley-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala:1048]: Why is this now a `SBuiltin`?

Comment (carlpulley-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala:1048]: Ah, managed to see why myself! `SBuiltinPure` is no longer required (as the `SpeedyHungry` stuff has now all gone).

Comment (nickchapman-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala:1048]: Sort of...

`SBuiltinPure` is a special case of `SBuiltin` that always returns a `Value` (and doesn't need to update any machine components)

In cases like `SBCastAnyContract`, which throw did an error, and now we want to return the error via the `Control`, I have to make them subclass directly from `SBuiltin`.

You could argue that the `Pure` tag was a bit of a lie!

**Review by remyhaemmerle-da — APPROVED** [review]
LGTM

**Pull request merged**